From 9d5e3dbd85d9a288aad2de349a33e0ad5a86ae98 Mon Sep 17 00:00:00 2001
From: Daniel Charles <daniel.charles@intel.com>
Date: Tue, 24 Mar 2015 11:37:59 -0700
Subject: [PATCH] check that intel-gen4asm tool is actually present

intel-gen4asm tool has to be present on the host tools so that
it can actually be used. Cross compiling shows problems when the
gpu-tools version is met but the tool is not available on the PATH

So, first check that the tool is available and later check for version
before defining HAVE_GEN4ASM.

Signed-off-by: Daniel Charles <daniel.charles@intel.com>
---
 configure.ac | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index a7d97e2..5a77020 100644
--- a/configure.ac
+++ b/configure.ac
@@ -78,8 +78,9 @@ AC_SUBST(LIBDRM_VERSION)
 
 dnl Check for gen4asm
 PKG_CHECK_MODULES(GEN4ASM, [intel-gen4asm >= 1.9], [gen4asm=yes], [gen4asm=no])
-AM_CONDITIONAL(HAVE_GEN4ASM, test x$gen4asm = xyes)
 AC_PATH_PROG([GEN4ASM], [intel-gen4asm])
+AS_IF([test "x$GEN4ASM" = "x" ], [AM_CONDITIONAL(HAVE_GEN4ASM, false)],
+    [AM_CONDITIONAL(HAVE_GEN4ASM, [test x$gen4asm = xyes])])
 
 dnl Check for git
 AC_ARG_VAR([GIT], [Path to git program, if any])
-- 
2.1.0

